src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,420 @@
+        var handler = new IoHandler(cts);
+
+        if (File.Exists(fileName))
+            File.Delete(fileName);
+
+        var stream = new FileStream(fileName, FileMode.CreateNew);
+        try
+        {
+            if (Program.Desktop.MainWindow is not null)
+            {
+                progressWindow = new IoProgressWindow {
+                    Title = $"Downloading {url}",
+                    Handler = handler
+                };
+                _ = progressWindow.ShowDialog(Program.Desktop.MainWindow);
+            }
+
+            using var httpClient = new HttpClient();
+            using HttpResponseMessage response =
+                await httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead, cts.Token);
+
+            await using Stream downloadStream = await response.Content.ReadAsStreamAsync(cts.Token);
+
+            if (response.Content.Headers.ContentLength is null)
+            {
+                handler.UpdateProgress(0, "Downloading (unknown length)");
+                await downloadStream.CopyToAsync(stream, cts.Token);
+            }
+            else
+            {
+                int read;
+                var length = (int)response.Content.Headers.ContentLength.Value;
+                var remaining = length;
+                while (!handler.Cancelled &&
+                       (read = await downloadStream.ReadAsync(
+                           buffer.AsMemory(0, int.Min(buffer.Length, remaining)),
+                           cts.Token)) >
+                       0)
+                {
+                    var prog = (float)read / length * 100;
+                    handler.UpdateProgress(prog, $"{float.Floor((1 - ((float)remaining / length)) * 100)}%");
+                    await stream.WriteAsync(buffer.AsMemory(0, read), cts.Token);
+                    remaining -= read;
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            if (ex is TaskCanceledException)
+                LogManager.GetCurrentClassLogger().Info("Download cancelled by user");
+            else
+                LogManager.GetCurrentClassLogger().Error(ex, "Failed to download!");
+
+            await stream.DisposeAsync();
+            return null;
+        }
+        finally
+        {
+            progressWindow?.Close();
+            stream.Close();
+        }
+
+        if (handler.Cancelled)
+            return null;
+
+        return stream;
+    }
+    /// <summary>
+    /// Returns the URL to the download link for the OS-specific version.
+    /// </summary>
+    /// <returns></returns>
+    private static string GetPath(GithubUpdate assets, OSPlatform OS)
+    {
+        var operatingSystem = "";
+
+        if (OS == OSPlatform.Windows)
+            operatingSystem = "win";
+        else if (OS == OSPlatform.Linux)
+            operatingSystem = "linux";
+        else
+            throw new Exception("Unsupported OS: " + nameof(OS));
```

Comment:
`nameof(OS)` here is just "OS", just `OS` is fine.

Also using a non-specific exception is a linter warning, use something like `ApplicationException`